modules/DNSWrapper.jsm

```diff
@@ -136,10 +154,13 @@ function asyncJSDNS_QueryDNS(name, rrtype) {
 			if (rcode !== undefined) {
 				result.rcode = rcode;
 			} else if (queryError !== undefined) {
-				result.rcode = RCODE.ServFail;
+				result.rcode = RCODE.ServFail;				
 			} else {
 				result.rcode = RCODE.NoError;
 			}
+			if (result.rcode != RCODE.NoError && queryError) {
+				throw new DKIM_InternalError(queryError, "DKIM_DNSERROR_SERVER_ERROR");
+			}
```

Comment:
Could you explain the reasoning behind this addition?

Don't see this hurting, but if you there trying to backport e57dff8a1bd352d29b3f5d383faf832d84ac709e here, I don't think you managed that. But I also don't see that as an important change to backport. That was only to avoid some possible confusion, see https://github.com/lieser/dkim_verifier/issues/231#issuecomment-1575194520.

Reply:
Well, I like the DNS errors in the log for diagnosing problems. This line will add the "queryError" string to the error exception to make it more visible. 
But the JSDNS behavior doesn't bother me in production, as I'm using libunbound to get DNSSEC features. Should I remove the exception again?

Reply:
You can leave it as is, was just trying to understand the motivation behind it.